test/12_dependency_versions/cibuildwheel_test.py

```diff
@@ -0,0 +1,115 @@
+import os
+import re
+import pytest
+import textwrap
+import cibuildwheel.util
+
+import utils
+
+VERSION_REGEX = r'([\w-]+)==([^\s]+)'
+
+
+def get_versions_from_constraint_file(constraint_file):
+    with open(constraint_file, encoding='utf8') as f:
+        constraint_file_text = f.read()
+
+    versions = {}
+
+    for package, version in re.findall(VERSION_REGEX, constraint_file_text):
+        versions[package] = version
+
+    return versions
+
+
+@pytest.mark.parametrize('python_version', ['2.7', '3.x'])
+def test_pinned_versions(python_version):
+    if utils.platform == 'linux':
+        pytest.skip('linux doesn\'t pin individual tool versions, it pins manylinux images instead')
+
+    project_dir = os.path.dirname(__file__)
+
+    build_environment = {}
+
+    if python_version == '2.7':
+        constraint_filename = 'constraints-python27.txt'
+    else:
+        constraint_filename = 'constraints.txt'
```

Comment:
There are more constraints files than these two? Why only this two? I see that in current release there are no difference `['pip', 'setuptools', 'wheel', 'virtualenv']` package version, but I expect at least some comment.